### PR TITLE
update k8s registry domain

### PR DESCRIPTION
**Pull request by cpanato**
**What type of PR is this?**
/kind cleanup

**Any specific area of the project related to this PR?**

/area plugins

**What this PR does / why we need it**:

- update k8s registry domain

part of https://github.com/kubernetes/k8s.io/issues/4780

/assign @leogr 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by leogr — COMMENTED** [review]
Since the rules can also run on environments not up-to-date, I'd keep both domains (ie `k8s.grc.io` and `registry.k8s.io`).

@LucaGuerra @darryk10 @loresuso wdyt?

**Comment by darryk10** [comment]
I agree on keeping them both. Some k8s version will still remain in the old domain so it makes sense to whitelist them both. In the future we would consider removing the old one.

**Comment by cpanato** [comment]
sound good and thanks for the feedback, i will update those and add a comment
@leogr @darryk10

File: plugins/k8saudit/rules/k8s_audit_rules.yaml
Comment (cpanato) [inline plugins/k8saudit/rules/k8s_audit_rules.yaml:189]: @leogr @darryk10 added this date to we remove the k8s reference after December 
the k8s registry will be frozen on 3rd of April 2023 (https://kubernetes.io/blog/2023/02/06/k8s-gcr-io-freeze-announcement/) so we give some extra time for people to update the clusters.
wdyt?

**Review by leogr — APPROVED** [review]
LGTM

/hold for a second opinion cc @falcosecurity/rules-maintainers @falcosecurity/plugins-maintainers

**Comment by LucaGuerra** [comment]
/unhold

LGTM, thanks!

**Pull request merged**
